test/analysis/typeComplexityAnalysis.test.ts

```diff
@@ -269,39 +301,482 @@ describe('Test getQueryTypeComplexity function', () => {
+                            weight: 1,
+                            fields: {},
+                        },
+                        human: {
+                            weight: 1,
+                            fields: {
+                                name: { weight: 0 },
+                                homePlanet: { weight: 0 },
+                                friends: {
+                                    resolveTo: 'character',
+                                    weight: mockCharacterFriendsFunction,
+                                },
+                                humanFriends: {
+                                    resolveTo: 'human',
+                                    weight: mockHumanFriendsFunction,
+                                },
+                            },
+                        },
+                        droid: {
+                            weight: 1,
+                            fields: {
+                                name: { weight: 0 },
+                                primaryFunction: { weight: 0 },
+                                friends: {
+                                    resolveTo: 'character',
+                                    weight: mockDroidFriendsFunction,
+                                },
+                            },
+                        },
+                    };
+                });
+                test('that have a complexity of zero', () => {
+                    query = `
+                        query {
+                            hero(episode: EMPIRE) {
+                                name
+                                ... on Droid {
+                                    primaryFunction
+                                }
+                                ... on Human {
+                                    homePlanet
+                                }
+                            }
+                        }`;
+                    // Query 1 + 1 hero
+                    expect(getQueryTypeComplexity(parse(query), variables, unionTypeWeights)).toBe(
+                        2
+                    );
+                });
+
+                test('that have differing complexities', () => {
+                    query = `
+                        query {
+                            hero(episode: EMPIRE) {
+                                name
+                                ... on Droid {
+                                    primaryFunction
+                                }
+                                ... on Human {
+                                    homePlanet
+                                    friends(first: 3) {
+                                        name
+                                    }
+                                }
+                            }
+                        }`;
+                    // Query 1 + 1 hero + max(Droid 0, Human 3) = 5
+                    mockHumanFriendsFunction.mockReturnValueOnce(3);
```

Comment:
The `friends` field on `human` invokes the `mockChracterFriendsFunction` not the `mockHumanFriendFunction`